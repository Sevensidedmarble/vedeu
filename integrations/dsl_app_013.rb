#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    height 10
    log '/tmp/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename:   '/tmp/dsl_app_013.out',
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: '/tmp/dsl_app_013.json'),
                # Vedeu::Renderers::HTML.new(filename: '/tmp/dsl_app_013.html'),
                # Vedeu::Renderers::Text.new(filename: '/tmp/dsl_app_013.txt'),
              ]
    run_once!
    standalone!
  end

  load File.dirname(__FILE__) + '/support/test_interface.rb'

  Vedeu.render do
    view(:test_interface) do
      lines do
        left 'view->lines->left 1', { foreground: '#ff0000' }
        left 'view->lines->left 2', { background: '#ff0000', foreground: '#000000' }
      end
    end
  end

  def self.actual
    File.read('/tmp/dsl_app_013.out')
  end

  def self.expected
    File.read(File.expand_path('../expected/dsl_app_013.out', __FILE__))
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start

if DSLApp.expected == DSLApp.actual
  puts "#{__FILE__} \e[32mPassed.\e[39m"
  exit 0;
else
  puts "#{__FILE__} \e[31mFailed.\e[39m"
  exit 1;
end
