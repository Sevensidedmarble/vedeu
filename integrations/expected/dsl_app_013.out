[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H [2;5HT[2;6He[2;7Hs[2;8Ht[2;9H [2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3H[38;5;196m[49mv[3;4Hi[3;5He[3;6Hw[3;7H-[3;8H>[3;9Hl[3;10Hi[3;11Hn[3;12He[3;13Hs[3;14H-[3;15H>[3;16Hl[3;17He[3;18Hf[3;19Ht[3;20H [3;21H1[3;22H [3;23H [3;24H [3;25H [3;26H [3;27H [3;28H [3;29H [3;30H [3;31H [3;32H[39m[49m(0x(B[0m[4;2H[39m[49m(0x(B[4;3H[38;5;16m[48;5;196mv[4;4Hi[4;5He[4;6Hw[4;7H-[4;8H>[4;9Hl[4;10Hi[4;11Hn[4;12He[4;13Hs[4;14H-[4;15H>[4;16Hl[4;17He[4;18Hf[4;19Ht[4;20H [4;21H2[4;22H [4;23H [4;24H [4;25H [4;26H [4;27H [4;28H [4;29H [4;30H [4;31H [4;32H[39m[49m(0x(B[0m[5;2H[39m[49m(0x(B[5;32H(0x(B[0m[6;2H[39m[49m(0m(B[6;3H(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H(0j(B[0m